darts/models/ensemble_model.py

```diff
@@ -24,33 +25,70 @@ class EnsembleModel(ForecastingModel):
+                                for model in models)
+        self.is_global_ensemble = all(isinstance(model, GlobalForecastingModel) for model in models)
+
+        raise_if_not(is_local_ensemble or self.is_global_ensemble,
+                     "All models must be instances of the same type, either darts.models.ForecastingModel"
+                     "or darts.models.GlobalForecastingModel",
                      logger)
         super().__init__()
-        self.models = models
+        self.models: Union[List[ForecastingModel], List[GlobalForecastingModel]] = models
 
-    def fit(self, series: TimeSeries) -> None:
+    def fit(self,
+            series: Union[TimeSeries, Sequence[TimeSeries]],
+            covariates: Optional[Union[TimeSeries, Sequence[TimeSeries]]] = None) -> None:
         """
         Fits the model on the provided series.
         Note that `EnsembleModel.fit()` does NOT call `fit()` on each of its constituent forecasting models.
         It is left to classes inheriting from EnsembleModel to do so appropriately when overriding `fit()`
         """
-        super().fit(series)
+        raise_if(not self.is_global_ensemble and not isinstance(series, TimeSeries),
+                 "All models are of type darts.models.ForecastingModel which do not support covariates.",
+                 logger
+                 )
+        raise_if(not self.is_global_ensemble and covariates is not None,
+                 "All models are of type darts.models.ForecastingModel which do not support covariates.",
+                 logger
+                 )
+        super().fit(series, covariates)
+
+    def _ts_sequence_to_multivariate_ts(self, ts_sequence: Sequence[TimeSeries]) -> TimeSeries:
+        if isinstance(ts_sequence, Sequence):
+            return reduce(lambda a, b: a.stack(b), ts_sequence)
+        else:
+            return ts_sequence
 
     def predict(self,
                 n: int,
-                num_samples: int = 1) -> TimeSeries:
-        super().predict(n, num_samples)
+                series: Optional[Union[TimeSeries, Sequence[TimeSeries]]] = None,
+                covariates: Optional[Union[TimeSeries, Sequence[TimeSeries]]] = None,
+                num_samples: int = 1,
+                ) -> Union[TimeSeries, Sequence[TimeSeries]]:
+
+        super().predict(n, series, covariates, num_samples)
+
+        if self.is_global_ensemble:
+            predictions = self._ts_sequence_to_multivariate_ts(
+                self.models[0].predict(n, series, covariates, num_samples))
+        else:
+            predictions = self.models[0].predict(n, num_samples)
 
-        predictions = self.models[0].predict(n, num_samples)
         if len(self.models) > 1:
             for model in self.models[1:]:
-                predictions = predictions.stack(model.predict(n, num_samples))
+                if self.is_global_ensemble:
+                    prediction = self._ts_sequence_to_multivariate_ts(
```

Comment:
Before we were stacking all the univariate forecasts of one target series into one TimeSeries because that could give us a convenient way to use regression models on this one TimeSeries. However now I think this aggregation should per-target-series and so for N targets you should end up with N predicted TimeSeries, each of which being the output of some ensembling logic, no?